pkg/scheduler/reconciliation/persistent.go

```diff
@@ -174,6 +160,40 @@ func (r *PersistentReconciliationRepository) RemoveReconciliation(schedulingID s
 	return db.Transaction(r.Conn, dbOps, r.Logger)
 }
 
+func (r *PersistentReconciliationRepository) RemoveReconciliations(schedulingIDs []string) error {
```

Comment:
Does it have to be in a transaction? I mean it's fine some of the delete batch failed, right? And put this bulk deletion in one transaction I believe it will have performance issue.

Reply:
For performance I have some screenshots attached above. Seeing which we decided on splitting the whole batch in blocks (sub-slices) of 200. 
Transaction enhancements are WIP and will be a part of upcoming PRs.